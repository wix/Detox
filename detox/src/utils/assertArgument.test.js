const assertions = require('./assertArgument');
const { assertUndefined } = require('./assertArgument');

describe('assertEnum', () => {
  const { assertEnum } = assertions;

  it('should create an assertion function for enums', () => {
    const assertSpeed = assertEnum(['fast', 'slow']);

    expect(() => assertSpeed({ speed: 'fast' })).not.toThrow();
    expect(() => assertSpeed({ speed: 'slow' })).not.toThrow();
    expect(() => assertSpeed({ speed: 'medium' })).toThrowErrorMatchingSnapshot();
  });
});

describe('assertNormalized', () => {
  const { assertNormalized } = assertions;

  it.each([
    0,
    0.5,
    1,
    NaN
  ])('should pass for %d', (validNumber) => {
    expect(() => assertNormalized({ validNumber })).not.toThrow();
  });

  it.each([
    -0.5,
    1.01,
    Infinity,
    null,
    undefined,
    '0.5'
  ])('should throw for %j', (invalidNumber) => {
    expect(() => assertNormalized({ invalidNumber })).toThrowErrorMatchingSnapshot();
  });
});

describe('assertNumber', () => {
  const { assertNumber } = assertions;

  it.each([
    42,
    NaN,
    Infinity,
    -Infinity,
  ])('should pass for %d', (validNumber) => {
    expect(() => assertNumber({ validNumber })).not.toThrow();
  });

  it.each([
    '42',
    false,
  ])('should throw for %j', (invalidNumber) => {
    expect(() => assertNumber({ invalidNumber })).toThrowErrorMatchingSnapshot();
  });
});

describe('assertString', () => {
  const { assertString } = assertions;

  it.each([
    '',
    '123',
  ])('should pass for %d', (validString) => {
    expect(() => assertString({ validString })).not.toThrow();
  });

  it.each([
    123,
    undefined,
  ])('should throw for %j', (invalidString) => {
    expect(() => assertString({ invalidString })).toThrowErrorMatchingSnapshot();
  });
});

describe('assertDuration', () => {
  const { assertDuration } = assertions;

  it.each([
    42,
    NaN,
    Infinity,
    -Infinity,
  ])('should pass for %d', (validNumber) => {
    expect(() => assertDuration(validNumber)).not.toThrow();
  });

  it.each([
    '42',
    false,
  ])('should throw for %j', (invalidNumber) => {
    expect(() => assertDuration(invalidNumber)).toThrowErrorMatchingSnapshot();
  });
});

describe('assertPoint', () => {
  const { assertPoint } = assertions;

  it('should pass for valid point', () => {
    expect(() => assertPoint({ x: 0, y: 0 })).not.toThrow();
  });

  it.each([
    { x: 0 },
    { y: 0 },
    { x: '0', y: 0 },
    { x: 0, y: '0' },
  ])('should throw for %j', (invalidPoint) => {
    expect(() => assertPoint(invalidPoint)).toThrowErrorMatchingSnapshot();
  });
});

describe('assertUndefined', () => {
  it('should pass for undefined', () => {
    expect(() => assertUndefined(undefined)).not.toThrow();
  });

  it.each([
    'str',
    1,
    { key: 'val' }
  ])('should throw for %j', (definedValue) => {
    expect(() => assertUndefined(definedValue)).toThrowErrorMatchingSnapshot();
  });
});
