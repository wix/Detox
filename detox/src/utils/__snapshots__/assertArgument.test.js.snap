// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertDuration should throw for "42" 1`] = `"duration should be a number, but got 42 (string)"`;

exports[`assertDuration should throw for false 1`] = `"duration should be a number, but got false (boolean)"`;

exports[`assertEnum should create an assertion function for enums 1`] = `"speed should be one of [fast, slow], but got medium (string)"`;

exports[`assertNormalized should throw for "0.5" 1`] = `"invalidNumber should be a number, but got 0.5 (string)"`;

exports[`assertNormalized should throw for -0.5 1`] = `"invalidNumber should be a number [0.0, 1.0], but got -0.5 (number)"`;

exports[`assertNormalized should throw for 1.01 1`] = `"invalidNumber should be a number [0.0, 1.0], but got 1.01 (number)"`;

exports[`assertNormalized should throw for null 1`] = `"invalidNumber should be a number [0.0, 1.0], but got Infinity (number)"`;

exports[`assertNormalized should throw for null 2`] = `"invalidNumber should be a number, but got null (object)"`;

exports[`assertNormalized should throw for undefined 1`] = `"invalidNumber should be a number, but got undefined (undefined)"`;

exports[`assertNumber should throw for "42" 1`] = `"invalidNumber should be a number, but got 42 (string)"`;

exports[`assertNumber should throw for false 1`] = `"invalidNumber should be a number, but got false (boolean)"`;

exports[`assertPoint should throw for {"x":"0","y":0} 1`] = `"point should be an object with x and y properties, but got {"x":"0","y":0}"`;

exports[`assertPoint should throw for {"x":0,"y":"0"} 1`] = `"point should be an object with x and y properties, but got {"x":0,"y":"0"}"`;

exports[`assertPoint should throw for {"x":0} 1`] = `"point should be an object with x and y properties, but got {"x":0}"`;

exports[`assertPoint should throw for {"y":0} 1`] = `"point should be an object with x and y properties, but got {"y":0}"`;

exports[`assertString should throw for 123 1`] = `"invalidString should be a string, but got 123 (number)"`;

exports[`assertString should throw for undefined 1`] = `"invalidString should be a string, but got undefined (undefined)"`;

exports[`assertUndefined should throw for "str" 1`] = `"0 expected to be undefined, but got s (string)"`;

exports[`assertUndefined should throw for {"key":"val"} 1`] = `"key expected to be undefined, but got val (string)"`;

exports[`assertUndefined should throw for 1 1`] = `"undefined is not iterable (cannot read property Symbol(Symbol.iterator))"`;
